Source/Shared/arcana/threading/cancellation.h

```diff
@@ -203,20 +233,24 @@ namespace arcana
             return { m_impl };
         }
 
-        void unsafe_cancel()
+        void cancel(bool block_until_cancellation_completed = false)
```

Comment:
nit: we seem to be using camelCase for variables and also remove `cancellation` since it's implied
```suggestion
        void cancel(bool blockUntilCompleted = false)
```